### PR TITLE
RCIAM-499_Extend_COmanage_Linking_Plugin

### PR DESCRIPTION
**Pull request by ioigoume**
- Add support for voPersonCertificate* attributes

### CODE REVIEW

File: Lib/CertUtils.php [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+   */
+  public static function getEnvValues($plg_cfg) {
+    // Get Certificate SDN and IDN mappings
+    $subject_dn_value = "";
+    $issuer_dn_value = "";
+    $subject_dn_map = !empty($plg_cfg['subject_dn_attribute']) ? $plg_cfg['subject_dn_attribute'] : null;
+    $issuer_dn_map = !empty($plg_cfg['issuer_dn_attribute']) ? $plg_cfg['issuer_dn_attribute'] : null;
+    // Get Subject DN env value if available
+    if(!empty($subject_dn_map)) {
+      $subject_dn_value = getenv($subject_dn_map);
```
Comment (lionick) [inline Lib/CertUtils.php]: is that equivalent to $subject_dn_value = (!empty(getenv($subject_dn_map)) ? getenv($subject_dn_map) : ""; 
?

File: Lib/CertUtils.php [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+      $subject_dn_value = !empty($subject_dn_value) ? $subject_dn_value : "";
+    }
+    // Get Issuer DN env value if available
+    if(!empty($issuer_dn_map)) {
+      $issuer_dn_value = getenv($issuer_dn_map);
```
Comment (lionick) [inline Lib/CertUtils.php]: is that equivalent to $issuer_dn_value = (!empty(getenv($issuer_dn_map)) ? getenv($issuer_dn_map) : "";

File: Lib/CertUtils.php [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+   */
+  public static function getEnvValuesFromCache($plg_cfg, $attribute_cache) {
+    // Get Certificate SDN and IDN mappings
+    $subject_dn_value = "";
+    $issuer_dn_value = "";
+    $subject_dn_map = !empty($plg_cfg['subject_dn_attribute']) ? $plg_cfg['subject_dn_attribute'] : null;
+    $issuer_dn_map = !empty($plg_cfg['issuer_dn_attribute']) ? $plg_cfg['issuer_dn_attribute'] : null;
+    // Get Subject DN env value if available
+    if(!empty($subject_dn_map)) {
+      $subject_dn_value = $attribute_cache[$subject_dn_map];
```
Comment (lionick) [inline Lib/CertUtils.php]: is that equivalent to $subject_dn_value = !empty($attribute_cache[$subject_dn_map]) ? $attribute_cache[$subject_dn_map] : "";
?

File: Lib/CertUtils.php [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+      $subject_dn_value = !empty($subject_dn_value) ? $subject_dn_value : "";
+    }
+    // Get Issuer DN env value if available
+    if(!empty($issuer_dn_map)) {
+      $issuer_dn_value = $attribute_cache[$issuer_dn_map];
```
Comment (lionick) [inline Lib/CertUtils.php]: is that equivalent to $issuer_dn_value = !empty($attribute_cache[$issuer_dn_map]) ? $attribute_cache[$issuer_dn_map] : "";
?

File: Model/LinkOrgIdentityEnroller.php [hunk 5]
```diff
@@ -454,7 +464,8 @@ public function findCoPersonforIdentifier($identifier, $co_id=null) {
   public function createOrgIdentity($registered_user, $cmp_attibutes_list, $email_verified, $plg_cfg) {
     $this->OrgIdentity = ClassRegistry::init('OrgIdentity');
 
-    // Create the data we need to save so as to create the OrgIdentity and all the relations
+    // Fetch Environmental Variables
+    // todo: Move this into its own function
```
Comment (lionick) [inline Model/LinkOrgIdentityEnroller.php]: this todo will remain?

File: Model/LinkOrgIdentityEnroller.php [hunk 6]
```diff
@@ -516,18 +527,35 @@ public function createOrgIdentity($registered_user, $cmp_attibutes_list, $email_
...
+        foreach($sdn_values as $dn) {
+          $association_data['Cert'][] = array(
+            'subject' => $dn,
+            'type' => CertEnum::X509,
+            'actor_identifier' => $cmp_attibutes_list[$user_id_attribute],
+          );
+        }
+      }
+    } else {
+      list($subject_value, $issuer_value) = CertUtils::getEnvValuesFromCache($plg_cfg, $cmp_attibutes_list);
```
Comment (lionick) [inline Model/LinkOrgIdentityEnroller.php]: list($subject_value, $issuer_value) = CertUtils::getEnvValuesFromCache($plg_cfg, $cmp_attibutes_list); 
can be put outside of if-else condition?

File: Model/LinkOrgIdentityEnroller.php [hunk 7]
```diff
@@ -516,18 +527,35 @@ public function createOrgIdentity($registered_user, $cmp_attibutes_list, $email_
...
-          'type'    => CertEnum::X509,
+          'subject' => trim($subject_value),
+          'issuer' => trim($issuer_value),
+          'type' => CertEnum::X509,
           'actor_identifier' => $cmp_attibutes_list[$user_id_attribute],
         ),
       );
     }
-  
+
```
Comment (lionick) [inline Model/LinkOrgIdentityEnroller.php:554]: This will stay? or is forgotten?

File: View/LinkOrgIdentityEnrollers/configure.ctp [hunk 8]
```diff
@@ -86,514 +86,594 @@ if (!$e) {
...
+    var confirm = $("#LinkOrgIdentityEnrollerEmailRedirectMode option:selected").val();
 
-        if(confirm != '<?php print LinkOrgIdentityRedirectModeEnum::Disabled; ?>') {
-            $("#LinkOrgIdentityEnrollerExpWindow").closest("ul.field-children").show('fade');
-            $("#LinkOrgIdentityEnrollerVerificationSubject").closest("li").show('fade');
-            $("#LinkOrgIdentityEnrollerVerificationBody").closest("li").show('fade');
-        } else {
-            $("#LinkOrgIdentityEnrollerExpWindow").closest("ul.field-children").hide('fade');
-        }
+    if (confirm != '<?php print LinkOrgIdentityRedirectModeEnum::Disabled; ?>') {
```
Comment (lionick) [inline View/LinkOrgIdentityEnrollers/configure.ctp]: remove space after "if" ?

**Review by lionick — CHANGES_REQUESTED** [review]

File: Model/LinkOrgIdentityEnroller.php (hunk 5 above)
Comment (ioigoume) [reply Model/LinkOrgIdentityEnroller.php]: no

File: View/LinkOrgIdentityEnrollers/configure.ctp (hunk 8 above)
Comment (ioigoume) [reply View/LinkOrgIdentityEnrollers/configure.ctp]: ok

File: Model/LinkOrgIdentityEnroller.php (hunk 6 above)
Comment (ioigoume) [reply Model/LinkOrgIdentityEnroller.php]: ok

File: Lib/CertUtils.php (hunk 4 above)
Comment (ioigoume) [reply Lib/CertUtils.php]: yes

File: Lib/CertUtils.php (hunk 3 above)
Comment (ioigoume) [reply Lib/CertUtils.php]: yes

File: Lib/CertUtils.php (hunk 2 above)
Comment (ioigoume) [reply Lib/CertUtils.php]: yes

File: Lib/CertUtils.php (hunk 1 above)
Comment (ioigoume) [reply Lib/CertUtils.php]: yes

File: Model/LinkOrgIdentityEnroller.php (hunk 7 above)
Comment (ioigoume) [reply Model/LinkOrgIdentityEnroller.php:554]: Removed redundant empty lines

**Pull request merged**